src/python_testing/TC_RR_1_1.py

```diff
@@ -0,0 +1,432 @@
+            sub = await client.ReadAttribute(nodeid=self.dut_node_id, attributes=subscription_contents,
+                                             reportInterval=(min_report_interval_sec, max_report_interval_sec), keepSubscriptions=False)
+            self._subscriptions.append(sub)
+
+            attribute_handler = AttributeChangeAccumulator(
+                name=client.name, expected_attribute=Clusters.Basic.Attributes.NodeLabel, output=output_queue)
+            sub.SetAttributeUpdateCallback(attribute_handler)
+            sub_handlers.append(attribute_handler)
+
+            # TODO: Replace resubscription catcher with API to disable re-subscription on failure
+            resub_catcher = ResubscriptionCatcher(name=client.name)
+            sub.SetResubscriptionAttemptedCallback(resub_catcher)
+            resub_catchers.append(resub_catcher)
+
+        asserts.assert_equal(len(self._subscriptions), len(client_list), "Must have the right number of subscriptions")
+
+        # Step 6: TODO: Read 9 paths and validate success
+        logging.info("Step 6: Read 9 paths (first 9 attributes of Basic Information cluster) and validate success")
+
+        large_read_contents = [
+            Clusters.Basic.Attributes.DataModelRevision,
+            Clusters.Basic.Attributes.VendorName,
+            Clusters.Basic.Attributes.VendorID,
+            Clusters.Basic.Attributes.ProductName,
+            Clusters.Basic.Attributes.ProductID,
+            Clusters.Basic.Attributes.NodeLabel,
+            Clusters.Basic.Attributes.Location,
+            Clusters.Basic.Attributes.HardwareVersion,
+            Clusters.Basic.Attributes.HardwareVersionString,
+        ]
+        large_read_paths = [(0, attrib) for attrib in large_read_contents]
+        basic_info = await dev_ctrl.ReadAttribute(self.dut_node_id, large_read_paths)
+
+        # Make sure everything came back from the read that we expected
+        asserts.assert_true(0 in basic_info.keys(), "Must have read endpoint 0 data")
+        asserts.assert_true(Clusters.Basic in basic_info[0].keys(), "Must have read Basic Information cluster data")
+        for attribute in large_read_contents:
+            asserts.assert_true(attribute in basic_info[0][Clusters.Basic],
+                                "Must have read back attribute %s" % (attribute.__name__))
+
+        # Step 7: Trigger a change on NodeLabel
+        logging.info(
+            "Step 7: Change attribute with one client, await all attributes changed successfully without loss of subscriptions")
+        await asyncio.sleep(1)
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=AFTER_LABEL))])
+
+        all_changes = {client.name: False for client in client_list}
+
+        # Await a stabilization delay in increments to let the event loops run
+        start_time = time.time()
+        elapsed = 0
+        time_remaining = timeout_delay_sec
+
+        while time_remaining > 0:
```

Comment:
I think testing for a single write getting reported feels like under testing here. Perhaps we could do this in a loop till the time_remaining hits?

Reply:
We are testing all writes come back from all clients. I am implementing the test plan as written.